profiles/bicycle.lua

```diff
@@ -338,7 +338,7 @@ function way_function (way, result)
     result.forward_mode = mode.train
     result.backward_mode = mode.train
     -- railways
-    if (not access and data.highway) or profile.access_tag_whitelist[access] then
+    if (not access and data.highway) or (access and not profile.access_tag_blacklist[access]) then
```

Comment:
Looking at https://github.com/Project-OSRM/osrm-backend/blob/cb32971a664ea97760173e06dbfd927779efc659/profiles/bicycle.lua#L286 the second clause should always be true since for blacklisted ways we don't get this far. I think the fix would be to not check access tags (which are meant for bikes, not trams) at all.